SignalMessaging/attachments/SendExternalFileViewController.m

```diff
@@ -176,6 +154,61 @@ - (UIView *)createTitleLabel
     return label;
 }
 
+#pragma mark - SelectThreadViewControllerDelegate
+
+- (void)threadWasSelected:(TSThread *)thread
+{
+    OWSAssert(self.attachment);
+    OWSAssert(thread);
+    self.thread = thread;
+
+    __weak typeof(self) weakSelf = self;
+
+    // FIXME SHARINGEXTENSION
+    // Handling safety number changes brings in a lot of machinery.
+    // How do we want to handle this?
+    // e.g. fingerprint scanning, etc. in the SAE or just redirect the user to the main app?
+    //    BOOL didShowSNAlert =
+    //        [SafetyNumberConfirmationAlert presentAlertIfNecessaryWithRecipientIds:thread.recipientIdentifiers
+    //                                                              confirmationText:[SafetyNumberStrings
+    //                                                              confirmSendButton]
+    //                                                               contactsManager:self.contactsManager
+    //                                                                    completion:^(BOOL didConfirm) {
+    //                                                                        if (didConfirm) {
+    //                                                                            [weakSelf threadWasSelected:thread];
+    //                                                                        }
+    //                                                                    }];
+    //    if (didShowSNAlert) {
+    //        return;
+    //    }
+
+    AttachmentApprovalViewController *approvalVC =
+        [[AttachmentApprovalViewController alloc] initWithAttachment:self.attachment delegate:self];
+
+    [self.navigationController pushViewController:approvalVC animated:YES];
+
+    // FIXME This implemenation was for the "import with signal" functionality,
+    // and is now broken. We need to be sure to remove the "import with signal" functionality.
+    // [SignalApp.sharedApp presentConversationForThread:thread];
+}
+
+#pragma mark - AttachmentApprovalViewControllerDelegate
+
+- (void)didApproveAttachment
+{
+    [ThreadUtil addThreadToProfileWhitelistIfEmptyContactThread:self.thread];
+    [ThreadUtil sendMessageWithAttachment:self.attachment inThread:self.thread messageSender:self.messageSender];
+
+    // FIXME SHARINGEXTENSION
+    // Show loading screen and dismiss entire share extnesion once entirely complete
+    [self.navigationController popViewControllerAnimated:YES];
```

Comment:
s/extnesion/extension/

Also, see `ShareViewController` which has a method for dismissing the whole share experience.  I think we need to "complete" the extension with an "item" in order to dismiss entire share UI as opposed to dismissing just our SAE.

Reply:
This sounds good, but I'm going to leave it for the next PR.